core/src/authentication/authentication.resolver.ts

```diff
@@ -1,16 +1,33 @@
 import {UseGuards} from "@nestjs/common";
-import {Mutation, Resolver} from "@nestjs/graphql";
+import {Args, Mutation, Resolver} from "@nestjs/graphql";
+
+import {OrganizationRepository} from "$repositories";
 
 import {AuthenticationService} from "./authentication.service";
-import {GraphQLJwtRefreshGuard} from "./strategies/jwt/guards";
+import {AuthenticatedUser} from "./decorators";
+import {GraphQLJwtAuthGuard, GraphQLJwtRefreshGuard} from "./strategies/jwt/guards";
+import {JwtAuthPayload, JwtRefreshPayload, JwtTokenSet} from "./types";
 
 @Resolver()
 export class AuthenticationResolver {
-    constructor(private readonly authenticationService: AuthenticationService) {}
+    constructor(
+        private readonly authenticationService: AuthenticationService,
+        private readonly organizationRepository: OrganizationRepository,
+    ) {}
 
-    @Mutation(() => String)
+    @Mutation(() => JwtTokenSet)
     @UseGuards(GraphQLJwtRefreshGuard)
-    async refreshLogin(): Promise<string> {
-        return "not implemented";
+    async refreshLogin(@AuthenticatedUser() user: JwtRefreshPayload): Promise<JwtTokenSet> {
+        return this.authenticationService.login(user.userId, user.currentOrganizationId);
+    }
+
+    @Mutation(() => JwtTokenSet)
+    @UseGuards(GraphQLJwtAuthGuard)
+    async switchOrganization(
+        @AuthenticatedUser() user: JwtAuthPayload,
+        @Args("organizationId") organizationId: number,
+    ): Promise<JwtTokenSet> {
+        const organization = await this.organizationRepository.getById(organizationId);
+        return this.authenticationService.login(user.userId, organization.id);
```

Comment:
Are we checking somewhere that the user is actually in the requested organization?

Reply:
Do we want to do that? This will in theory allow us to have an application/form to non members to request membership or something. Will refactor if we want to explicitly say "this person is a member".

Reply:
It looks like it actually does happen in login; but yeah I would say that we do want that. If we want to request membership that can be a user guard, not a member guard.

If any user can switch to any organization I could see that causing issues elsewhere